routes/api/v1/favorites.js

```diff
@@ -18,4 +24,97 @@ router.post('/', (request, response) => {
     });
 });
 
+router.get('/', (request, response) => {
+  // console.log(request.body.api_key)
+  database('users').where('api_key', request.body.api_key)
+
+////////////////////////////////////////////////////////////////////////////////
+  .then(user => {
+    // console.log(user[0].id)
+    return database('favorites').where( 'user_id', user[0].id)
+  })
+
+////////////////////////////////////////////////////////////////////////////////
+  .then((favorites) => {
+    favorites_array = favorites.map(function (favorite_object) {
+    console.log(favorite_object.city)
+    return favorite_object.city
+});
+console.log(favorites_array)
+return(favorites_array)
+})
+
+////////////////////////////////////////////////////////////////////////////////
+.then((array) => {
+array.forEach((city) => {
+  var options = {
+      uri: 'https://maps.googleapis.com/maps/api/geocode/json',
+      qs: {
+          key: process.env.GOOGLE_API_KEY,
+          address: city
+      },
+      json: true
+  };
+
+rp(options).then(body => {
+    latitude = body.results[0].geometry.location.lat;
+    longitude = body.results[0].geometry.location.lng;
+    return body.results[0].geometry.location
+})
+
+
+////////////////////////////////////////////////////////////////////////////////
+.then((whatIsThis) => {
+    // officialResponse = "lalaal";
+    var options2 = {
+        uri: `https://api.darksky.net/forecast/${process.env.DARK_SKY_API_KEY}/${latitude},${longitude}?exclude=minutely`,
+        json: true
+    };
+
+    rp(options2)
+
+////////////////////////////////////////////////////////////////////////////////
+    .then(body2 => {
+      var weatherHash = {}
+      weatherHash["location"] = city;
+      weatherHash["currently"] = body2.currently
+
+      delete weatherHash.currently.time
```

Comment:
@fentontaylor could you take a look to see how I can better organize this code into different parts/folders? Also, what do you think of my approach to deleting things from the incoming JSON hash rather than save the data into a POJO first, kind of like what we were doing in mod 3